apps/spruce/src/pages/version/Tabs/index.tsx

```diff
@@ -184,67 +182,63 @@ export const VersionTabs: React.FC<VersionTabProps> = ({ version }) => {
       numFailedChildPatches,
       numStartedChildPatches,
       numSuccessChildPatches,
-      versionId: versionId ?? "",
+      versionId: version.id,
     });
-  }, [taskCount, childPatches, versionId]);
+  }, [taskCount, childPatches, version.id]);
 
   const activeTabs = useMemo(
-    // @ts-expect-error: FIXME. This comment was added by an automated script.
-    () => Object.keys(allTabs).filter((t) => tabIsActive[t] as PatchTab[]),
+    () =>
+      (Object.keys(allTabs) as VersionPageTabs[]).filter((t) => tabIsActive[t]),
     [allTabs, tabIsActive],
   );
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  const isValidTab = tabIsActive[tab];
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  const [selectedTab, setSelectedTab] = useState(activeTabs.indexOf(tab));
-  const previousTab = usePrevious(selectedTab);
+  const [selectedTab, setSelectedTab] = useState(tab || VersionPageTabs.Tasks);
 
-  useEffect(() => {
-    // If tab is not valid, set to task tab.
-    if (!isValidTab) {
-      // @ts-expect-error: FIXME. This comment was added by an automated script.
-      navigate(getVersionRoute(versionId), { replace: true });
+  const handleTabChange = (
+    newTab: VersionPageTabs,
+    sendAnalytics: boolean = false,
+  ) => {
+    if (!tabIsActive[newTab]) {
+      return;
     }
-    // If tab updates in URL without having clicked a tab (e.g. clicked build variant), update state here.
-    // @ts-expect-error: FIXME. This comment was added by an automated script.
-    else if (selectedTab !== activeTabs.indexOf(tab)) {
-      // @ts-expect-error: FIXME. This comment was added by an automated script.
-      setSelectedTab(activeTabs.indexOf(tab));
-    }
-  }, [tab]); // eslint-disable-line react-hooks/exhaustive-deps
-
-  // Update the URL and selectedTab state based on new tab selected.
-  const selectNewTab = (newTabIndex: number) => {
     const queryParams = parseQueryString(search);
-    const newTab = activeTabs[newTabIndex];
-    // @ts-expect-error: FIXME. This comment was added by an automated script.
-    const newRoute = getVersionRoute(versionId, {
-      tab: newTab as PatchTab,
-      ...queryParams,
-    });
-    navigate(newRoute, { replace: true });
 
-    if (previousTab !== undefined && previousTab !== newTabIndex) {
-      sendEvent({
-        name: "Changed tab",
-        tab: newTab as PatchTab,
-      });
+    setSelectedTab(newTab);
+    // In cases where we're changing tabs due to a non user action (e.g. a redirect we want to avoid sending analytics)
+    if (sendAnalytics) {
+      sendEvent({ name: "Changed tab", tab: newTab });
```

Comment:
does this ever get called? It seems like whenever we're calling the `handleTabChange` function in this file, `sendAnalytics` is false

Reply:
Thanks fixed!